File: anchor/programs/gigentic/src/states/escrow.rs
use anchor_lang::prelude::*;

#[account]
#[derive(InitSpace)]
pub struct Escrow {
    pub customer: Pubkey, // the customer who will pay the money to the escrow
    pub service_provider: Pubkey, // the service provider who will receive the money from the escrow
    pub fee_percentage: u8, // the fee percentage that the service provider will pay to the registry
    pub expected_amount: u64, // the expected amount for the service
    pub fee_account: Pubkey, // the fee account that will receive the fee from the service provider
}

File: web/lib/types/escrow.ts
import { BN } from '@coral-xyz/anchor';
import { PublicKey } from '@solana/web3.js';

export interface EscrowAccount {
  publicKey: PublicKey;
  account: {
    customer: PublicKey;
    serviceProvider: PublicKey;
    expectedAmount: BN;
    feeAccount: PublicKey;
    feePercentage: number;
  };
}
